packages/core/src/messages/acknowledgement.ts

```diff
@@ -0,0 +1,75 @@
+import { Challenge } from './challenge'
+import { deriveAckKeyShare, PublicKey } from '@hoprnet/hopr-utils'
+import { ecdsaSign, ecdsaVerify, publicKeyCreate } from 'secp256k1'
+import { SECRET_LENGTH, HASH_ALGORITHM } from './constants'
+import { SECP256K1_CONSTANTS, u8aSplit, CryptoError } from '@hoprnet/hopr-utils'
+import { createHash } from 'crypto'
+import type PeerId from 'peer-id'
+
+export class Acknowledgement {
```

Comment:
@peterbraden we use this in both core and core-ethereum right? maybe we move it into utils

Reply:
Afaik this is not used in core-ethereum. Core-ethereum used a class that is similarly named but different purpose

Reply:
regardless, let's move all 'serialization' classes together into utils.

Reply:
agreed

Reply:
@peterbraden why are you suggesting to migrate the serialization classes to `utils`?

Reply:
that's where the others all live. 

Utils is a bad name at the moment, it's more like 'lib'

Reply:
Postponing this because it creates chaos in utils